be/src/storage/data_dir.cpp

```diff
@@ -298,7 +298,22 @@ Status DataDir::load() {
         }
         return true;
     };
-    Status load_tablet_status = TabletMetaManager::walk(_kv_store, load_tablet_func);
+    Status load_tablet_status =
+            TabletMetaManager::walk_with_limit_time(_kv_store, load_tablet_func, config::load_tablet_limit_seconds);
+    if (load_tablet_status.is_time_out()) {
+        Status s = _kv_store->compact();
+        if (!s.ok()) {
+            LOG(ERROR) << "data dir " << _path << " compact meta befor load failed";
+            return s;
+        }
+        for (auto tablet_id : tablet_ids) {
+            _tablet_manager->drop_tablet(tablet_id, kKeepMetaAndFiles);
```

Comment:
is this operation risky？

Reply:
> is this operation risky？

This operation is only remove tablet in memory but not remove data file and meta file in disks, so i think this operation is safe.